docs/bigtable-data-api.rst

```diff
@@ -0,0 +1,328 @@
+
+Cells vs. Columns vs. Column Families
++++++++++++++++++++++++++++++++++++++
+
+* As we saw before, a table can have many column families.
+* As we'll see below, a table also has many rows (specified by row keys).
+* Within a row, data is stored in a cell. A cell simply has a value (as
+  bytes) and a timestamp. The number of cells in each row can be
+  different, depending on what was stored in each row.
+* Each cell lies in a column (**not** a column family). A column is really
+  just a more **specific** modifier within a column family. A column
+  can be present in every column family, in only one or anywhere in between.
+* Within a column family there can be many columns. For example within
+  the column family ``foo`` we could have columns ``bar`` and ``baz``.
+  These would typically be represented as ``foo:bar`` and ``foo:baz``.
+
+Modifying Data
+++++++++++++++
+
+Since data is stored in cells, which are stored in rows, the
+:class:`Row <gcloud.bigtable.row.Row>` class is the only class used to
+modify (write, update, delete) data in a
+:class:`Table <gcloud.bigtable.table.Table>`.
+
+Row Factory
+-----------
+
+To create a :class:`Row <gcloud.bigtable.row.Row>` object
+
+.. code:: python
+
+    row = table.row(row_key)
+
+Unlike the previous string values we've used before, the row key must
+be ``bytes``.
+
+Direct vs. Conditional vs. Append
+---------------------------------
+
+There are three ways to modify data in a table, described by the
+`MutateRow`_, `CheckAndMutateRow`_ and `ReadModifyWriteRow`_ API
+methods.
+
+* The **direct** way is via `MutateRow`_ which involves simply
+  adding, overwriting or deleting cells.
+* The **conditional** way is via `CheckAndMutateRow`_. This method
+  first checks if some filter is matched in a a given row, then
+  applies one of two sets of mutations, depending on if a match
+  occurred or not. (These mutations sets are called the "true
+  mutations" and "false mutations".)
+* The **append** way is via `ReadModifyWriteRow`_. This simply
+  appends (as bytes) or increments (as an integer) data in a presumed
+  existing cell in a row.
+
+Building Up Mutations
+---------------------
+
+In all three cases, a set of mutations (or two sets) are built up
+on a :class:`Row <gcloud.bigtable.row.Row>` before they are sent of
+in a batch via :meth:`commit() <gcloud.bigtable.row.Row.commit>`:
+
+.. code:: python
+
+    row.commit()
+
+To send **append** mutations in batch, use
+:meth:`commit_modifications() <gcloud.bigtable.row.Row.commit_modifications>`:
+
+.. code:: python
+
+    row.commit_modifications()
+
+We have a small set of methods on the :class:`Row <gcloud.bigtable.row.Row>`
+to build these mutations up.
+
+Direct Mutations
+----------------
+
+Direct mutations can be added via one of four methods
+
+* :meth:`set_cell() <gcloud.bigtable.row.Row.set_cell>` allows a
+  single value to be written to a column
+
+  .. code:: python
+
+      row.set_cell(column_family_id, column, value,
```

Comment:
Yeah that is how Bigtable works. HBase / Thrift combines them into a single value `fam:col` but I wanted to stay true-ish to the API in this implementation.